server/tests/unit/services/RunService.test.js

```diff
@@ -224,18 +224,421 @@ describe('RunService', () => {
+                const apgExample = await db.ApgExample.findOne({
+                    where: { test_version_id: testVersion.id }
+                });
+
+                const activeRuns = await RunService.configureRuns({
+                    test_version_id: testVersion.id,
+                    apg_example_ids: [apgExample.id],
+                    at_browser_pairs: [
+                        {
+                            at_name_id: at.at_name_id,
+                            at_version: atVersionNumber2,
+                            browser_id: browser.id,
+                            browser_version: browserVersionNumber2
+                        }
+                    ],
+                    run_status_id: runStatus.id
+                });
+
+                // Check the returned data
+                const keys = Object.keys(activeRuns);
+                const runId = parseInt(keys[0]);
+                expect(keys.length).toEqual(1);
+                expect(activeRuns[runId]).toEqual({
+                    id: runId,
+                    browser_id: browser.id,
+                    browser_version: browserVersionNumber2,
+                    browser_name: browser.name,
+                    at_id: at.id,
+                    at_key: at.key,
+                    at_name: at.AtName.name,
+                    at_version: atVersionNumber2,
+                    apg_example_directory: apgExample.directory,
+                    apg_example_name: apgExample.name,
+                    apg_example_id: apgExample.id,
+                    run_status_id: runStatus.id,
+                    run_status: db.RunStatus.RAW,
+                    test_version_id: testVersion.id,
+                    testers: []
+                });
+
+                // Check that previously activated pair is now inactive
+                await techPairDeactivate.reload();
+                expect(techPairDeactivate.active).toBe(false);
+
+                const activeBrowserVersionToAtVersions = await db.BrowserVersionToAtVersion.findAll(
+                    { where: { active: true } }
+                );
+                expect(activeBrowserVersionToAtVersions.length).toBe(1);
+
+                // Verify that new browser/at pair was created
+                const activeCreatedBrowserVersion = await db.BrowserVersion.findAll(
+                    {
+                        where: {
+                            id:
+                                activeBrowserVersionToAtVersions[0]
+                                    .browser_version_id
+                        }
+                    }
+                );
+                expect(activeCreatedBrowserVersion[0].version).toBe(
+                    browserVersionNumber2
+                );
+
+                const activeCreatedAtVersion = await db.AtVersion.findAll({
+                    where: {
+                        id: activeBrowserVersionToAtVersions[0].at_version_id
+                    }
+                });
```

Comment:
If you are looking for one object just use `findOne` or in this case `findByPk`.

And in this case, as you are looking for a related association you should be preloading it with `include` and then then referencing it directly like:

 ```
                const activeBrowserVersionToAtVersions = await db.BrowserVersionToAtVersion.findAll(
                    { where: { active: true } },
                    { include: [db.AtVersion]}
                );
activeBrowserVersionToAtVersions.AtVersion
```

Reply:
Thank you! Good catch